presto-parquet/src/main/java/io/prestosql/parquet/writer/ParquetWriter.java

```diff
@@ -0,0 +1,251 @@
+        }
+        rows += page.getPositionCount();
+
+        if (bufferedBytes >= writerOption.getMaxBlockSize()) {
+            columnWriters.forEach(ColumnWriter::close);
+            flush();
+            columnWriters.forEach(ColumnWriter::reset);
+            rows = 0;
+        }
+    }
+
+    @Override
+    public void close()
+            throws IOException
+    {
+        if (closed) {
+            return;
+        }
+        closed = true;
+        columnWriters.forEach(ColumnWriter::close);
+
+        flush();
+        writeFooter();
+        outputStream.close();
+    }
+
+    // Parquet File Layout:
+    //
+    // MAGIC
+    // variable: Data
+    // variable: Metadata
+    // 4 bytes: MetadataLength
+    // MAGIC
+    private void flush()
+            throws IOException
+    {
+        // write header
+        if (!writeHeader) {
+            createDataOutput(MAGIC).writeData(outputStream);
+            writeHeader = true;
+        }
+
+        // get all data in buffer
+        ImmutableList.Builder<BufferData> builder = ImmutableList.builder();
+        for (ColumnWriter columnWriter : columnWriters) {
+            columnWriter.getBuffer().forEach(builder::add);
+        }
+        List<BufferData> bufferDataList = builder.build();
+
+        // update stats
+        long stripeStartOffset = outputStream.size();
+        List<ColumnMetaData> metadatas = bufferDataList.stream()
+                .map(BufferData::getMetaData)
+                .collect(toImmutableList());
+        updateRowGroups(updateColumnMetadataOffset(metadatas, stripeStartOffset));
+
+        // flush pages
+        bufferDataList.stream()
+                .map(BufferData::getData)
+                .flatMap(List::stream)
+                .forEach(data -> data.writeData(outputStream));
+    }
+
+    private void writeFooter()
+    {
+        checkState(closed);
+        Slice footer = getFooter(rowGroupBuilder.build(), messageType);
+        createDataOutput(footer).writeData(outputStream);
+
+        Slice footerSize = Slices.allocate(SIZE_OF_INT);
+        footerSize.setInt(0, footer.length());
+        createDataOutput(footerSize).writeData(outputStream);
+
+        createDataOutput(MAGIC).writeData(outputStream);
+    }
+
+    static Slice getFooter(List<RowGroup> rowGroups, MessageType messageType)
+    {
+        FileMetaData fileMetaData = new FileMetaData();
+        fileMetaData.setVersion(1);
+        fileMetaData.setSchema(MessageTypeConverter.toParquetSchema(messageType));
+        long totalRows = rowGroups.stream().mapToLong(RowGroup::getNum_rows).sum();
+        fileMetaData.setNum_rows(totalRows);
+        fileMetaData.setRow_groups(ImmutableList.copyOf(rowGroups));
+
+        DynamicSliceOutput dynamicSliceOutput = new DynamicSliceOutput(40);
+        try {
+            Util.writeFileMetaData(fileMetaData, dynamicSliceOutput);
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
```

Comment:
Actually we don't need to use `UncheckedIOException` here.  The call chain is `close` -> `writeFooter`  -> `getFooter`, and `close` throws `IOEexception`